src/Orleans.Persistence.CosmosDB/CosmosDBGrainStorage.cs

```diff
@@ -71,6 +72,14 @@ public class CosmosDBGrainStorage : IGrainStorage, ILifecycleParticipant<ISiloLi
                 { CLEAR_STATE_SPROC, $"{CLEAR_STATE_SPROC}.js" },
                 { LOOKUP_INDEX_SPROC, $"{LOOKUP_INDEX_SPROC}.js" }
             };
+
+            if (_options.JsonSerializerSettings == null)
+            {
+                _options.JsonSerializerSettings = OrleansJsonSerializer.UpdateSerializerSettings(OrleansJsonSerializer.GetDefaultSerializerSettings(_typeResolver, _grainFactory),
+                    _options.UseFullAssemblyNames,
+                    _options.IndentJson,
+                    _options.TypeNameHandling);

```

Comment:
Make sure all instance fields/properties/methods are qualified with _this._ across the code.